.github/workflows/deploy_artifacts.yaml

```diff
@@ -0,0 +1,111 @@
+        required: false
+        type: boolean
+        default: false
+      quayRepository:
+        description: The quay repository to upload the image to. The repository must belong to the dockstore quay organization.
+        required: false
+        type: string
+      dockerContext:
+        description: The Docker context containing the Dockerfile of the image to build and push.
+        required: false
+        default: .
+        type: string
+
+env:
+  IS_DEVELOP_SNAPSHOT: ${{ github.ref_type == 'branch' && github.ref_name == 'develop' }}
+
+jobs:
+  deploy_maven:
+    name: Maven deploy ${{ github.ref_type == 'tag' && 'tagged' || 'snapshot' }} release
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+      # Step that does that actual cache save and restore
+      - uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+           ${{ runner.os }}-maven-
+
+      - name: install git secrets
+        run: |
+          wget --no-verbose -O git-secrets-1.3.0.tar.gz https://github.com/awslabs/git-secrets/archive/1.3.0.tar.gz
+          tar -zxf git-secrets-1.3.0.tar.gz
+          cd git-secrets-1.3.0
+          sudo make install   
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21.0.2+13.0.LTS'
+          distribution: 'adopt'
+          # settings.xml configuration
+          server-id: ${{ github.ref_type == 'tag' && 'central' || 'snapshots' }}
+          server-username: DEPLOY_USERNAME
+          server-password: DEPLOY_TOKEN
+
+      - name: Set changelist version
+        run: |
+          set -x
+          if ${{ github.ref_type == 'tag' }}; then
+            # Check that the tag follows semantic versioning. Note that the regex is not super strict
+            if [[ ${{ github.ref_name }} =~ ^[0-9]+\.[0-9]+\.[0-9a-zA-Z-]+(-[0-9a-zA-Z-]+\.[0-9a-zA-Z-]+)*$ ]]; then
+              # Break up the semantic version tag by the '.' delimiter and return the third field onward. Prefix this with the '.'
+              # Ex: 1.16.0 -> .0 and 1.16.0-alpha.0 -> .0-alpha.0
+              CHANGELIST_VERSION=.$(echo ${{ github.ref_name }} | cut -d. -f 3-)
+            else
+              echo "Invalid semantic version for tag ${{ github.ref_name }}"
+              exit 1
+            fi
+          elif ${{ github.ref_name != 'develop' }}; then
+            CHANGELIST_VERSION=.0-${{ github.ref_name }}-SNAPSHOT
+            CHANGELIST_VERSION=${CHANGELIST_VERSION//\//-}
+          fi
+
+          echo "CHANGELIST_VERSION=${CHANGELIST_VERSION}" >> $GITHUB_ENV
+
+      - name: Store Maven project version 
+        run: |
+          set -x
+          if ${{ env.IS_DEVELOP_SNAPSHOT }}; then
+            echo "maven_project_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
```

Comment:
Should this be `mvnw` instead of `mvn`? It probably doesn't matter for such a simple command, but maybe Maven isn't on the path at some point, etc., so might as well. Looks like you rely on `.mvnw` being present further below, on lines 97 and 99, so it seems safe.